Account/FxADeviceRegistrator.swift

```diff
@@ -0,0 +1,125 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Deferred
+import Shared
+
+private let log = Logger.syncLogger
+
+// The current version of the device registration, we use this to re-register
+// devices after we update what we send on device registration.
+private let DeviceRegistrationVersion = 1
+
+public enum FxADeviceRegistrationResult {
+    case RegisteredOrUpdated
+    case AlreadyRegistered
+}
+
+public class FxADeviceRegistrator {
+    public enum DeviceRegistratorError: MaybeErrorType {
+        case DeviceRegistrationFailed
+
+        public var description: String {
+            switch self {
+            case DeviceRegistrationFailed: return "Device registration failed."
+            }
+        }
+    }
+
+    public static func registerOrUpdateDevice(account: FirefoxAccount, state: MarriedState) -> Deferred<Maybe<FxADeviceRegistrationResult>> {
```

Comment:
Does it make sense to have these methods be static? I wonder if it would be easier to have them as regular methods and pass in the `FxAClient10` object as an init parameter (dependency injection). This might make unit testing this easier since you can pass in a mock client.